src/game/Entities/Vehicle.cpp

```diff
@@ -303,16 +303,17 @@ void VehicleInfo::Board(Unit* passenger, uint8 seat)
     auto* creatureDisplayInfo = sCreatureDisplayInfoStore.LookupEntry(static_cast<Creature*>(m_owner)->GetNativeDisplayId());
     float scale = creatureDisplayInfo->scale;
     scale *= sCreatureModelDataStore.LookupEntry(creatureDisplayInfo->ModelId)->Scale;
-    for (auto& attachment : sModelAttachmentStore[creatureDisplayInfo->ModelId])
-    {
-        if (attachment.id == attachmentLookup(seatEntry->m_attachmentID))
+    if (sModelAttachmentStore.find(creatureDisplayInfo->ModelId) != sModelAttachmentStore.end())
```

Comment:
Assign it to an itr and use that itr now that you already looked it up

Reply:
fair enough